core/src/test/java/org/hibernate/ogm/backendtck/batchfetching/MultiGetEmbeddedIdTest.java

```diff
@@ -52,32 +52,38 @@
 			EmptyOptionsContext.INSTANCE );
 
-	private static final EntityKeyMetadata METADATA = new DefaultEntityKeyMetadata( "BoardGame", new String[] { "id.name", "id.publisher" } );
+	private static final EntityKeyMetadata METADATA = new DefaultEntityKeyMetadata( "BoardGame", new String[]{ "id.name", "id.publisher" } );
 
-	private static final EntityKey NOT_IN_THE_DB = new EntityKey( METADATA, new Object[] { "none", "none" } );
+	private static final EntityKey NOT_IN_THE_DB = new EntityKey( METADATA, new Object[]{ "none", "none" } );
 	private static final BoardGame DOMINION = new BoardGame( "Rio Grande Games", "Dominion" );
 	private static final BoardGame KING_OF_TOKYO = new BoardGame( "Fantasmagoria", "King of Tokyo" );
 	private static final BoardGame SPLENDOR = new BoardGame( "Space Cowboys", "Splendor" );
 
 	@Test
 	public void testGetTuplesWithoutNulls() throws Exception {
-		try (OgmSession session = openSession()) {
-			session.getTransaction().begin();
-			MultigetGridDialect dialect = multiGetGridDialect();
+		try ( OgmSession session = openSession() ) {
+			Transaction tx = session.beginTransaction();
+			try {
+				MultigetGridDialect dialect = multiGetGridDialect();
 
-			EntityKey[] keys = new EntityKey[] { key( SPLENDOR ), key( DOMINION ), key( KING_OF_TOKYO ) };
-			List<Tuple> tuples = dialect.getTuples( keys, TUPLECONTEXT );
+				EntityKey[] keys = new EntityKey[]{ key( SPLENDOR ), key( DOMINION ), key( KING_OF_TOKYO ) };
+				List<Tuple> tuples = dialect.getTuples( keys, TUPLECONTEXT );
 
-			assertThat( tuples.get( 0 ).get( "id.publisher" ) ).isEqualTo( SPLENDOR.getId().getPublisher() );
-			assertThat( tuples.get( 0 ).get( "id.name" ) ).isEqualTo( SPLENDOR.getId().getName() );
+				assertThat( tuples.get( 0 ).get( "id.publisher" ) ).isEqualTo( SPLENDOR.getId().getPublisher() );
+				assertThat( tuples.get( 0 ).get( "id.name" ) ).isEqualTo( SPLENDOR.getId().getName() );
 
-			assertThat( tuples.get( 1 ).get( "id.publisher" ) ).isEqualTo( DOMINION.getId().getPublisher() );
-			assertThat( tuples.get( 1 ).get( "id.name" ) ).isEqualTo( DOMINION.getId().getName() );
+				assertThat( tuples.get( 1 ).get( "id.publisher" ) ).isEqualTo( DOMINION.getId().getPublisher() );
+				assertThat( tuples.get( 1 ).get( "id.name" ) ).isEqualTo( DOMINION.getId().getName() );
 
-			assertThat( tuples.get( 2 ).get( "id.publisher" ) ).isEqualTo( KING_OF_TOKYO.getId().getPublisher() );
-			assertThat( tuples.get( 2 ).get( "id.name" ) ).isEqualTo( KING_OF_TOKYO.getId().getName() );
+				assertThat( tuples.get( 2 ).get( "id.publisher" ) ).isEqualTo( KING_OF_TOKYO.getId().getPublisher() );
+				assertThat( tuples.get( 2 ).get( "id.name" ) ).isEqualTo( KING_OF_TOKYO.getId().getName() );
 
-			session.getTransaction().commit();
+				session.getTransaction().commit();
+			}
+			catch (Exception e) {
+				rollback( tx );
+				throw e;
```

Comment:
not critical as it's just a test, but keep in mind that:
 - you could have an exception thrown during commit()
 - it's illegal to try to rollback a transaction which was committed (even if it failed to commit)